packages/cli/src/cmds/beacon/handler.ts

```diff
@@ -156,8 +156,15 @@ export async function beaconHandlerInit(args: BeaconArgs & GlobalArgs) {
   const {peerId, enr} = await initPeerIdAndEnr(args, beaconPaths.beaconDir, logger);
   // Inject ENR to beacon options
   beaconNodeOptions.set({network: {discv5: {enr: enr.encodeTxt(), config: {enrUpdate: !enr.ip && !enr.ip6}}}});
-  // Add simple version string for libp2p agent version
-  beaconNodeOptions.set({network: {version: version.split("/")[0]}});
+
+  if (args.private) {
+    beaconNodeOptions.set({network: {private: true}});
+  } else {
+    // Add simple version string for libp2p agent version
+    beaconNodeOptions.set({network: {version: version.split("/")[0]}});
+    // Casting to `as string` so that if version type changes this line does not compile
```

Comment:
Curious about the comment, isn't the point of using types to catch those issue. I mean this line should not compile if it is broken because we changed some type of `version`. 

As per typescript compiler, it does not even complain if `version` is an object. Good news is we catch this now via eslint rule added in https://github.com/ChainSafe/lodestar/pull/5562.

Reply:
Indeed, with https://github.com/ChainSafe/lodestar/pull/5562 it's not necessary. It's annoying tho how type relaxed typescript is

Reply:
I guess typescript is just lax here because of how template literals work in javascript, but it's definitely not ideal